### ISSUE TITLE
SessionTimeout

### ISSUE THREAD

**Issue by SOF3** [deotern/JamAuth#5]
Why does SessionTimeout extend `pocketmine\scheduler\Task`? It should extend PluginTask. The plugin should not do anything after being disabled according to PocketMine standards.

P.S. Why not continue with HereAuth? You can build a forum account provider using HereAuth easily :thinking:

**Comment by NIIP99** [comment]
Well, idk. At the moment, I'm just thinking of more memory is required to save JamAuth to parent class. So, extend to Task would consume less memory, maybe? There are a bunch of SessionTimeout will be called anyway.

I'm not that familiar with HereAuth. And another reason is I wanted the development of JamAuth can be independent and constant, not required to wait for other dev to do the job.
Plus, JamAuth is intended to extend to java, for Nukkit. So, it's better not to depend on other plugins. :)

**Comment by SOF3** [comment]
Extending Task instead of PluginTask saves 32 or 64 bits. Do you think it is helpful? (Excuses!) (It wastes less memory than any smallest premature optimization anywhere in this plugin!)

A plugin is considered to have disappeared from the server forever after it is disabled. Using Task instead of PluginTask is virtually **always** wrong for plugins.

About HereAuth, it might be helpful if you do not create yet another auth plugin to further complicate the confusion for users (look at all those 🍴 ...), or for other plugins that want to integrate with auth plugins. At least, HereAuth is currently almost best-featured, and I can't think of anything that you want to do with HereAuth but cannot do so through extension plugins for it. If you are worried about it being not maintained, actually, everyone can fork it and continue working on it if it needs any maintenance.

**Comment by SOF3** [comment]
BTW, don't rely on __destruct of a session. It is unreliable when PHP destructs the object.

**Comment by SOF3** [comment]
Not gonna create another issue, but I would like to remind you that it is not a good idea to import a database on the main thread.

**Comment by NIIP99** [comment]
Well, thanks for your guide.
I will constantly improving JamAuth

Anyway, I will still prefer develop the plugin by myself as I planned to extend to java. I don't wanted to mess things up :)

Yep, I do know. I've tried with Async before. But I required to echo the progress messages (Importing: 14%) to console. And I found no way AsyncTask can do this. Maybe there is? But I don't know as PocketMine API can't be accessed.
